include/envoy/stats/sink.h

```diff
@@ -43,10 +43,8 @@ class MetricSnapshot {
 
   /**
    * @return the time in milliseconds when the snapshot was created.
-   *
-   * Stores converted timestamp here to avoid repeated conversions for every metric.
    */
-  virtual int64_t snapshotTimeMs() const PURE;
+  virtual std::chrono::milliseconds snapshotTime() const PURE;
```

Comment:
Why not just use `SystemTime` here? My having the interface return the data as milliseconds I think you might be dropping precision, and some call-site in the future might want microseconds or something.

Reply:
Not sure if any one wants microseconds for this though. If you think that is useful, I can change

Reply:
+1, that would make it more clear what we are talking about, though if SystemTime what time zone is that? I think this whole thing needs a bit more detail.

/wait

Reply:
SystemTime is I think in UTC and you would convert to a time-zone when printing?

Reply:
OK I wasn't sure of SystemTime is guaranteed to be UTC. Either way I agree we should use it and also make sure the API docs specify that the result is time since UTC epoch?

/wait

Reply:
Ok. Moved to SystemTime. PTAL